### PR TITLE
Creating Grid for Checkbox Images

### PR DESCRIPTION
**Pull request by korneliakobiela**
[Issue] #1325
[Problem] Checkbox image grid bad styling
[Solution] Add proper styles for new Guideline

Signed-off-by: Kornelia Kobiela <korneliak.95@gmail.com>

### CODE REVIEW

File: src/js/profile/mobile/widget/GridView.js [hunk 1]
```diff
@@ -149,7 +149,13 @@
 					 * @style ui-gridview-handler
 					 * @member ns.widget.mobile.GridView
 					 */
-					HANDLER: "ui-gridview-handler"
+					HANDLER: "ui-gridview-handler",
+					/**
+					 * Set image as checked
+					 * @style ui-element-checked
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/GridView.js]: I think better will be `ui-gridview-image-checked`.

File: src/js/profile/mobile/widget/GridView.js [hunk 2]
```diff
@@ -299,6 +307,7 @@
 
 				self.on("animationend webkitAnimationEnd", animationEndCallback);
 
+				self.on("change", self._shadeCheckbox);
 				utilsEvents.on(page, pageEvents.SHOW, self._onSetGridStyle);
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/GridView.js:311]: Please add `self.off("change", self._shadeCheckbox);` in method `_unbindEvents`

File: src/js/profile/mobile/widget/GridView.js [hunk 3]
```diff
@@ -622,6 +631,35 @@
...
+					self._shadeCheckbox(checkboxNodeList[i]);
+				}
+			}
+
+			/**
+			 * Set darker background for checked element in the grid
+			 * @method _shadeCheckbox
+			 * @protected
+			 * @member ns.widget.mobile.GridView
+			 */
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/GridView.js:662]: `@param {HTMLElement} target`

File: src/js/profile/mobile/widget/GridView.js [hunk 4]
```diff
@@ -299,6 +307,7 @@
 
 				self.on("animationend webkitAnimationEnd", animationEndCallback);
 
+				self.on("change", self._shadeCheckbox);
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/GridView.js]: You can assign this listener to `self` and extend method `handleEvent`

File: src/js/profile/mobile/widget/GridView.js [hunk 5]
```diff
@@ -622,6 +631,35 @@
...
+			 * @method _initCheckboxState
+			 * @protected
+			 * @member ns.widget.mobile.GridView
+			 */
+			prototype._initCheckboxState = function (element) {
+				var checkboxNodeList = element.querySelectorAll("input[type=checkbox]"),
+					i = 0,
+					self = this;
+
+				for (i = 0; i < checkboxNodeList.length; i++) {
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/GridView.js]: good practice is create local `length` variable for `array.length` property in loop.

File: src/js/profile/mobile/widget/GridView.js (hunk 4 above)
Comment (korneliakobiela) [reply src/js/profile/mobile/widget/GridView.js]: I've tried to do it but `handleEvent` function doesn't seem to be called.

File: src/js/profile/mobile/widget/GridView.js (hunk 2 above)
Comment (korneliakobiela) [reply src/js/profile/mobile/widget/GridView.js:311]: Done in LINE334

File: src/js/profile/mobile/widget/GridView.js (hunk 3 above)
Comment (korneliakobiela) [reply src/js/profile/mobile/widget/GridView.js:662]: Done

File: src/js/profile/mobile/widget/GridView.js (hunk 4 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/mobile/widget/GridView.js]: This is impossible that the method is not calling.

`self.on("change", self);`

and add case to handleEvent:
```
case "change":
           self._shadeCheckbox(event.target);
       break;
```

Comment (korneliakobiela) [reply src/js/profile/mobile/widget/GridView.js]: Thanks for the clarification. Done

File: src/js/profile/mobile/widget/GridView.js [hunk 6]
```diff
@@ -370,6 +380,9 @@
 				var self = this;
 
 				switch (event.type) {
+					case "change":
+						self._shadeCheckbox(event);
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/GridView.js]: You don't need all event object... You need only `event.target` or not?

**Review by TomaszLukawskiSam — APPROVED** [review]
Looks good to me.

**Pull request merged**